tests/spectron/spectronWebActions.js

```diff
@@ -27,16 +29,140 @@ class WebActions {
         })
     }
 
+    async closeWindowByClick() {
+        await this.app.client.click(ui.CLOSE_BUTTON);
+    }
+
     async openApplicationMenuByClick() {
         await this.app.client.click(ui.MAIN_MENU_ITEM);
     }
-    
-    async closeWindowByClick() {        
-        await this.app.client.click(ui.CLOSE_BUTTON);
+
+    async  getElementByXPath(xpath) {
+        var elem = this.app.client.element(xpath);
+        if (elem.isVisible()) {
+            return elem;
+        }
+        return null;
+    }
+    async  inputText(el, data) {
+        var obj = await this.getElementByXPath(el);
+        if (obj != null)
+            await this.app.client.setValue(el, data);
     }
+    async clickAndWaitElementVisible(xpath, elementToVisible, timeOut = constants.TIMEOUT_WAIT_ELEMENT) {
+        await this.app.client.click(xpath).then(async () => {
+            await this.app.client.waitForVisible(elementToVisible, timeOut);
 
-    async openApplicationMenuByClick() {       
-        await this.app.client.click(ui.MAIN_MENU_ITEM);
+        });
+    }
+
+    async login(username) {
+        await this.inputText(ui.SIGN_IN_EMAIL, username);
+        await this.inputText(ui.SIGN_IN_PASSWORD, process.env.PASSWORD);
```

Comment:
let's use DRY principles, we can have `process.env.PASSWORD` in a variable called `SIGN_IN_PASSWORD` and use that across the file.